connect-inject/health_check_handler.go

```diff
@@ -0,0 +1,317 @@
+package connectinject
+
+// handler skeleton inspired by https://github.com/trstringer/k8s-controller-core-resource
+
+import (
+	ctx "context"
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/cenkalti/backoff"
+	"github.com/hashicorp/consul-k8s/subcommand/flags"
+	"github.com/hashicorp/consul/api"
+	log "github.com/hashicorp/go-hclog"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// HCHandler interface contains the methods that are required
+type HCHandler interface {
+	Init() error
+	ObjectCreated(obj interface{}) error
+	ObjectDeleted(obj interface{}) error
+	ObjectUpdated(objNew interface{}) error
+	Reconcile() error
+}
+
+const (
+	healthCheckPassing  = "passing"
+	healthCheckCritical = "critical"
+)
+
+// HealthCheckHandler is a sample implementation of HCHandler
+type HealthCheckHandler struct {
+	Log                log.Logger
+	AclConfig          api.NamespaceACLConfig
+	Client             *api.Client
+	HFlags             *flags.HTTPFlags
+	Clientset          kubernetes.Interface
+	ConsulClientScheme string
+	ConsulPort         string
+}
+
+// getConsulHealthCheckID deterministically generates a health check ID that will be unique to the Agent
+// where the health check is registered and deregistered.
+func (t *HealthCheckHandler) getConsulHealthCheckID(pod *corev1.Pod) string {
+	return fmt.Sprintf("%s_%s_kubernetes-health-check-ttl", pod.Namespace, t.getConsulServiceID(pod))
+}
+
+// return the serviceID of the connect service
+func (t *HealthCheckHandler) getConsulServiceID(pod *corev1.Pod) string {
```

Comment:
Can this be a shared function between the connect inject code so we use the same logic for determining service id?